.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,42 @@
+name: Docker Publish
+
+on:
+  push:
+    branches: [ main ]
```

Comment:
Guessing we do not want to build for pull requests?

Reply:
I didn't think so. What I cribbed from ran some tests and built but not published the images. I could re-add the last bit back in if we want

Reply:
`push: ${{ github.event_name != 'pull_request' }}` should take care of that. I suspect building the image on PRs since it is one of the main artifacts, might not be a bad thing.

Reply:
899baf008271025ac6d652e1bfd433ee00de6756 which is different from what was in the cribbed workflow as that seemed to be generating a syntax error

Reply:
Which has now built this branch. So this is self-proving?